annotation/annotation/tasks/services.py

```diff
@@ -62,6 +61,37 @@
     ValidationSchema,
 )
 
+
+class LRU:
+    def __init__(self, capacity):
+        self.capacity = capacity
+        self.counter = 0
+        self.cache = {}
+        self.lru = {}
+
+    def __contains__(self, item):
+        if item in self.cache:
+            return True
+        else:
+            return False
+
+    def __getitem__(self, key):
+        if key in self.cache:
+            self.lru[key] = self.counter
+            self.counter += 1
+            return self.cache[key]
+        raise KeyError
+
+    def __setitem__(self, key, value):
+        if len(self.cache) >= self.capacity:
+            old_key = min(self.lru.keys(), key=lambda k: self.lru[k])
```

Comment:
The implementation isn't efficient enough. Try to avoid using the min function and aim for insert and delete operations to be in O(1).

Reply:
Due to the suggestion, I will rewrite the implementation using `OrderedDict` since it has O(1) methods for insert and delete, as it is a double linked list.